harp/reader.py

```diff
@@ -44,13 +45,37 @@ def __init__(
         self.read = read
 
 
+class RegisterMap(UserDict[str, RegisterReader]):
+    _address_map: dict[int, str]
+
+    def __init__(self, registers: dict[str, RegisterReader]) -> None:
+        super().__init__(registers)
+        self._address_map = {self[key].register.address: key for key in self.keys()}
```

Comment:
I believe we can avoid the lookup here by directly iterating through the map values instead of the keys:

```suggestion
        self._address_map = {value.register.address: value for value in registers.values()}
```